src/sys/statfs.rs

```diff
@@ -1,20 +1,665 @@
+        self.0.f_ffree
+    }
+
+    /// Free file nodes in filesystem
+    #[cfg(all(
+        target_pointer_width = "32",
+        not(any(
+            target_os = "ios",
+            target_os = "macos",
+            target_os = "android",
+            target_os = "freebsd",
+            target_os = "openbsd",
+            target_os = "dragonfly",
+            all(
+                target_os = "linux",
+                target_env = "musl"
+            )
+        ))
+    ))]
+    pub fn files_free(&self) -> libc::uint32_t {
+        self.0.f_ffree
+    }
+
+    /// Free file nodes in filesystem
+    #[cfg(any(
+        target_os = "freebsd",
+        target_os = "openbsd"
+    ))]
+    pub fn files_free(&self) -> i64 {
+        self.0.f_ffree
+    }
+
+    /// Free file nodes in filesystem
+    #[cfg(target_os = "dragonfly")]
+    pub fn files_free(&self) -> libc::c_long {
+        self.0.f_ffree
+    }
+
+    /// Filesystem ID
+    pub fn filesystem_id(&self) -> fsid_t {
+        self.0.f_fsid
+    }
+}
 
-        Errno::result(res).map(drop)
+impl Debug for Statfs {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("Statfs")
+            .field("filesystem_type", &self.filesystem_type())
+            .field("optimal_transfer_size", &self.optimal_transfer_size())
+            .field("block_size", &self.block_size())
+            .field("blocks", &self.blocks())
+            .field("blocks_free", &self.blocks_free())
+            .field("blocks_available", &self.blocks_available())
+            .field("files", &self.files())
+            .field("files_free", &self.files_free())
+            .field("filesystem_id", &self.filesystem_id())
+            .finish()
     }
 }
 
-pub fn fstatfs<T: AsRawFd>(fd: &T, stat: &mut libc::statfs) -> Result<()> {
+pub fn statfs<P: ?Sized + NixPath>(path: &P) -> Result<Statfs> {
     unsafe {
-        Errno::clear();
-        Errno::result(libc::fstatfs(fd.as_raw_fd(), stat)).map(drop)
+        let mut stat: Statfs = mem::uninitialized();
+        let res = path.with_nix_path(|path| libc::statfs(path.as_ptr(), &mut stat.0))?;
+        Errno::result(res).map(|_| stat)
+    }
+}
+
+pub fn fstatfs<T: AsRawFd>(fd: &T) -> Result<Statfs> {
+    unsafe {
+        let mut stat: Statfs = mem::uninitialized();
+        Errno::result(libc::fstatfs(fd.as_raw_fd(), &mut stat.0)).map(|_| stat)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::fs::File;
+
+    use sys::statfs::*;
+    use sys::statvfs::*;
+
+    #[test]
+    fn statfs_call() {
+        check_statfs("/tmp");
+        check_statfs("/dev");
+        check_statfs("/run");
+        check_statfs("/");
+    }
+
+    #[test]
+    fn fstatfs_call() {
+        check_fstatfs("/tmp");
+        check_fstatfs("/dev");
+        check_fstatfs("/run");
+        check_fstatfs("/");
+    }
+
+    fn check_fstatfs(path: &str) {
+        let vfs = statvfs(path.as_bytes());
+        if vfs.is_err() {
```

Comment:
You shouldn't ignore all errors from `statvfs` and friends.  Better to check that the path exists and then assert on error.